File: scripts/solve.js
(async function() {

	let args        = process.argv.splice(2);
	let argSolution = args[0] ?? null;
	let argInput    = args[1] ?? null;
	
	let yearID;
	let dayID;
	
	if ( typeof argSolution === 'string' && argSolution.match(/^\d+$/g ) ) {
		yearID = '2024'; // default to most recent year
		dayID  = argSolution.padStart(2, '0');
	} else if ( typeof argSolution === 'string' && argSolution.match(/^\d+-\d+$/g ) ) {
		let parts = argSolution.split('-');
		yearID = parts[0];
		dayID  = parts[1].padStart(2, '0');
	} else {
		console.error('Please provide a day number using the --day flag, e.g. `npm run solve 1` or `npm run solve 2022-1`.');
		return;
	}
	
	const path    = process.cwd() + `/solutions/${yearID}/${dayID}`;
	const inputID = argInput ?? 'input';
	
	const { Solution } = await import(`${path}/solution.js`);
	const instance = new Solution( path, inputID );
	
	console.info( `${yearID} DAY ${dayID}` );
	console.info( `-----------` );
	
	instance.init();
	
})();

File: solve.py
import os
import sys
from importlib.machinery import SourceFileLoader

def init():

	yearID  = sys.argv[1]          if len(sys.argv) >= 2 else '2025'
	dayID   = sys.argv[2].zfill(2) if len(sys.argv) >= 3 else '01'
	inputID = sys.argv[3]          if len(sys.argv) >= 4 else 'input'

	module_path = os.path.join( 'solutions', f'y{yearID}', f'd{dayID}' )
	module_file = os.path.join( module_path, 'solution.py' )
	solution = SourceFileLoader('solution', module_file ).load_module()

	title = f'{yearID} » DAY {dayID} » {inputID}'
	sys.stdout.write( title + '\n' + ( '-' * len(title) ) + '\n' );

	solution.init( module_path, inputID )

init()